inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java

```diff
@@ -17,66 +17,180 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(SQLServerSource.class);
+    private static final Integer DEBEZIUM_QUEUE_SIZE = 100;
+    private ExecutorService executor;
+    public InstanceProfile profile;
+    private BlockingQueue<SourceData> debeziumQueue;
+    private final Properties props = new Properties();
+    private String dbName;
+    private String schemaName;
+    private String tableName;
 
     public SQLServerSource() {
     }
 
-    @Override
-    public List<Reader> split(TaskProfile conf) {
-        SQLServerReader sqlServerReader = new SQLServerReader();
-        List<Reader> readerList = new ArrayList<>();
-        readerList.add(sqlServerReader);
-        sourceMetric.sourceSuccessCount.incrementAndGet();
-        return readerList;
+    protected void initSource(InstanceProfile profile) {
+        try {
+            LOGGER.info("SQLServerSource init: {}", profile.toJsonStr());
+            debeziumQueue = new LinkedBlockingQueue<>(DEBEZIUM_QUEUE_SIZE);
+
+            dbName = profile.get(TaskConstants.TASK_SQLSERVER_DBNAME);
+            schemaName = profile.get(TaskConstants.TASK_SQLSERVER_SCHEMA_NAME);
+            tableName = profile.get(TaskConstants.TASK_SQLSERVER_TABLE_NAME);
+
+            props.setProperty("name", "SQLServer-" + instanceId);
+            props.setProperty("connector.class", SqlServerConnector.class.getName());
+            props.setProperty("offset.storage", FileOffsetBackingStore.class.getName());
+            String agentPath = AgentConfiguration.getAgentConf()
+                    .get(AgentConstants.AGENT_HOME, AgentConstants.DEFAULT_AGENT_HOME);
+            String offsetPath = agentPath + "/" + getThreadName() + "/offset.dat";
+            props.setProperty("offset.storage.file.filename", offsetPath);
+            props.setProperty("offset.flush.interval.ms", "10000");
+            props.setProperty("database.history", FileDatabaseHistory.class.getCanonicalName());
+            props.setProperty("database.history.file.filename", agentPath + "/" + getThreadName() + "/history.dat");
+            // ignore "schema" and extract data from "payload"
+            props.setProperty("key.converter.schemas.enable", "false");
+            props.setProperty("value.converter.schemas.enable", "false");
+            // ignore ddl
+            props.setProperty("include.schema.changes", "false");
+            // convert time to formatted string
+            props.setProperty("converters", "datetime");
+            props.setProperty("datetime.type", "org.apache.inlong.agent.plugin.utils.SQLServerTimeConverter");
+            props.setProperty("datetime.format.date", "yyyy-MM-dd");
+            props.setProperty("datetime.format.time", "HH:mm:ss");
+            props.setProperty("datetime.format.datetime", "yyyy-MM-dd HH:mm:ss");
+            props.setProperty("datetime.format.timestamp", "yyyy-MM-dd HH:mm:ss");
```

Comment:
Why do we need to make special adjustments to the time format? Should we keep it as it is

Reply:
The time data from debezium is timestamp. If not make special adjustments, sort-flink can't deserialize it success

Reply:
> timestamp

Confirmed, sortlink can parse timestamps

Reply:
DATE or DATETIME data coming from Debezium sql server connector are long type. But the Flink sort can't parse unix timestamp, only SQL and ISO-8601. So we should convert the time to "yyyy-MM-dd HH:mm:ss" and let flink sort can parse them